src/auth/OIDCConnector/OIDCSecured.tsx

```diff
@@ -202,6 +202,46 @@ export function OIDCSecured({
     setCookie(auth.user?.access_token ?? '', auth.user?.expires_at ?? 0);
   }, [auth]);
 
+  useEffect(() => {
+    let visibilityState = document.visibilityState;
+    function loginOnTokenExpired() {
+      if (authRef.current.user?.expires_at) {
+        const currentDate = new Date();
+        // KC does not give the correct timestamp UNIX format, it has to be multiplied by 1000
+        const expiredDate = new Date(authRef.current.user?.expires_at * 1000);
+        if (expiredDate < currentDate) {
+          state.login();
+        }
+      }
+    }
+    function visibilityListener() {
+      if (visibilityState !== document.visibilityState && document.visibilityState === 'visible') {
+        // we went from non visible to visible (tab went into focus for various reasons)
+        loginOnTokenExpired();
+      }
+      visibilityState = document.visibilityState;
+    }
+
+    function networkOnlineListener() {
+      loginOnTokenExpired();
+      authRef.current.startSilentRenew();
```

Comment:
@Hyperkid123 could we add some basic tests to confirm these actions are performed when the event listeners are fired? This is one of those areas we could regress without noticing for awhile.

Otherwise approach looks good to me.

Reply:
I'll also likely need to test this in our itless env

Reply:
This will be very hard to test without a custom KC instance 😢 or we would have to wait for token expiration, turn off then network etc.

I can do basic testing that the pause/resume functionality works, but timing out the tokens might be difficult. I think I should be able to mock it, but this would need a proper e2e testing.

Reply:
Would it be possible to have a stubbed version of KC that has functionality for deliberately expiring tokens for the sake of testing? Seems like that's something we could use in future tests, so there would be future ROI on the initial work.

Reply:
We could have a separate e2e test and run the KC instance locally and decrease the timeouts, yes. But we do not have an environment for that right now. Once we have some e2e available in Konflux we can do that.

Reply:
@Hyperkid123 getting back to this PR - I get we will need to wait for E2E - but would it be possible to move these function definitions outside of the `useEffect` and test them with jest tests? I'd like to make sure all the expected functions are called and we consider any undefined/falsy objects.

Reply:
Sure. But I think we should use cypress instead. JSDOM events are most likely just fake.

Reply:
@florkbr added the tests.